### PR TITLE
Refactor Web3 Signer + Bug fixes

### PR DESCRIPTION
**Pull request by FrederikBolding**
- [x] Initial refactor
- [x] Fix bug with requesting signing after unmount
- [x] Fix issue with changing networks

### CODE REVIEW

**Comment by FrederikBolding** [comment]
[ch6391]

File: src/components/SignTransactionWallets/Web3.tsx [hunk 1]
```diff
@@ -1,193 +1,90 @@
...
 import translate, { translateRaw } from '@translations';
 import { ISignComponentProps, TAddress } from '@types';
-import { getWeb3Config, isSameAddress, withHook } from '@utils';
+import { getWeb3Config, isSameAddress } from '@utils';
 
 import './Web3.scss';
 
 enum WalletSigningState {
-  READY, //when signerWallet is ready to sendTransaction
+  //READY, //when signerWallet is ready to sendTransaction
```
Comment (GuillaumeRx) [inline src/components/SignTransactionWallets/Web3.tsx:15]: If it's not used can we remove it ?

**Comment by blurpesec** [comment]
@FrederikBolding can you fix this issue here too? :)
image

**Review by blurpesec — APPROVED** [review]
Other than the alignment of the message, this looks good to me! I tested MM tx signing on desktop. For release testing, we should make sure this works fine on mobile as well

**Pull request merged**